### PR TITLE
constexpr compile errors with ICC 2019

### PR DESCRIPTION
**Pull request by rkowalewski**
- `constexpr` lambdas are not allowed before C++17, see this [proposal](http://open-std.org/JTC1/SC22/WG21/docs/papers/2015/n4487.pdf)
- `constexpr` of `RegionSpec` makes no sense since not all members can be constructed by a `constexpr`

### CODE REVIEW

File: dash/test/memory/GlobStaticMemTest.cc [hunk 1]
```diff
@@ -178,7 +178,7 @@ TEST_F(GlobStaticMemTest, MakeUniqueMoveSemantics)
   constexpr size_t lcapA = 20;
   constexpr size_t lcapB = lcapA / 2;
 
-  constexpr auto cap_in_bytes = [](size_t lcap) {
+  auto cap_in_bytes = [](size_t lcap) {
```
Comment (devreal) [inline dash/test/memory/GlobStaticMemTest.cc]: I guess `const` would make sense here though

Comment (rkowalewski) [reply dash/test/memory/GlobStaticMemTest.cc]: All right, done.

**Comment by rkowalewski** [comment]
Can we merge this? I need this fix. ;)

**Comment by pascalj** [comment]
Yes, this also resolves issues with the current GCC.

**Review by pascalj — APPROVED** [review]
LGTM

**Pull request merged**